File: simulators/de.unistuttgart.iste.sqa.mpw.hamstersimulator.java/ui/src/main/java/de/unistuttgart/hamstersimulator/examples/Example01.java
package de.unistuttgart.hamstersimulator.examples;

class Example01 extends SimpleHamsterGame {

    /**
     * How often Paule should repeat its behaviour.
     */
    private static final int SCENARIO_REPETITION_COUNT = 20;

    @Override
    protected void run() {
        displayInNewGameWindow();
        for (int i = 0; i < SCENARIO_REPETITION_COUNT; i++) {
            paule.write("Hallo!");
            paule.move();
            paule.move();
            paule.pickGrain();
            paule.pickGrain();
            paule.putGrain();
            paule.putGrain();
            paule.turnLeft();
            paule.turnLeft();
            paule.move();
            paule.move();
            paule.turnLeft();
            paule.turnLeft();
        }
    }

}

File: simulators/de.unistuttgart.iste.sqa.mpw.hamstersimulator.cpp/ui/examples/Example01.cpp
#include <HamsterCore.h>
#include <HamsterView.h>

#include "SdlApplication.h"
#include "HamsterApplicationHandler.h"
#include <string>

const int SCREEN_WIDTH = 640;
const int SCREEN_HEIGHT = 480;

using namespace hamstersimulator;
using namespace hamster;

void hamsterProgram() {

}

int main() {
    auto game = HamsterGame::create();
    TerritoryLoader::initializeFor(*game)->loadFromResourceFile("resources/territories/example01.ter");
    
    HamsterApplicationHandler applicationHandler(game, [&]()
    {
        auto paule = game->getTerritory()->getDefaultHamster();

        int a = paule->readNumber("Hallo nummer");
        auto text = paule->readString("Hallo string");
        paule->write("Hallo! " + std::to_string(a) + ", " + text);

        paule->move();
        paule->move();
        paule->pickGrain();
        paule->turnLeft();
        paule->turnLeft();
        paule->move();
        paule->move();
        paule->putGrain();
        paule->turnLeft();
        paule->turnLeft();
        paule->move();
    });

    SdlApplication application("Example 01", applicationHandler);
    application.initialize(SCREEN_WIDTH, SCREEN_HEIGHT);

    application.runApplication();

    return 0;
}

